src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/UserService.kt

```diff
@@ -127,6 +130,50 @@ class UserService(
     }
   }
 
+  fun updateUserRolesAndQualifications(id: UUID, userRolesAndQualifications: UserRolesAndQualifications): AuthorisableActionResult<UserEntity> {
+    val user = userRepository.findByIdOrNull(id) ?: return AuthorisableActionResult.NotFound()
+    return updateUserRolesAndQualificationsToUserEntity(user, userRolesAndQualifications)
+  }
+
+  fun updateUserRolesAndQualificationsToUserEntity(user: UserEntity, userRolesAndQualifications: UserRolesAndQualifications): AuthorisableActionResult<UserEntity> {
+    val roles = userRolesAndQualifications.roles
+    val qualifications = userRolesAndQualifications.qualifications
+
+    clearQualifications(user)
+    clearRolesForService(user, ServiceName.approvedPremises)
+
+    roles.forEach {
+      this.addRoleToUser(user, transformUserRoles(it))
+    }
+
+    qualifications.forEach {
+      this.addQualificationToUser(user, transformQualifications(it))
+    }
+
+    return AuthorisableActionResult.Success(user)
+  }
+
+  fun transformQualifications(qualification: APIUserQualification): UserQualification = when (qualification) {
+    APIUserQualification.emergency -> UserQualification.EMERGENCY
+    APIUserQualification.esap -> UserQualification.ESAP
+    APIUserQualification.lao -> UserQualification.LAO
+    APIUserQualification.pipe -> UserQualification.PIPE
+    APIUserQualification.womens -> UserQualification.WOMENS
+  }
+
+  fun transformUserRoles(approvedPremisesUserRole: ApprovedPremisesUserRole): UserRole = when (approvedPremisesUserRole) {
+    ApprovedPremisesUserRole.assessor -> UserRole.CAS1_ASSESSOR
+    ApprovedPremisesUserRole.matcher -> UserRole.CAS1_MATCHER
+    ApprovedPremisesUserRole.manager -> UserRole.CAS1_MANAGER
+    ApprovedPremisesUserRole.workflowManager -> UserRole.CAS1_WORKFLOW_MANAGER
+    ApprovedPremisesUserRole.applicant -> UserRole.CAS1_APPLICANT
+    ApprovedPremisesUserRole.roleAdmin -> UserRole.CAS1_ADMIN
+    ApprovedPremisesUserRole.reportViewer -> UserRole.CAS1_REPORT_VIEWER
+    ApprovedPremisesUserRole.excludedFromAssessAllocation -> UserRole.CAS1_EXCLUDED_FROM_ASSESS_ALLOCATION
+    ApprovedPremisesUserRole.excludedFromMatchAllocation -> UserRole.CAS1_EXCLUDED_FROM_MATCH_ALLOCATION
+    ApprovedPremisesUserRole.excludedFromPlacementApplicationAllocation -> UserRole.CAS1_EXCLUDED_FROM_PLACEMENT_APPLICATION_ALLOCATION
+  }
```

Comment:
Thinking about this, I do wonder if we could put this in somewhere (maybe the `util` folder) so we can improve the test coverage and allow it to be used elsewhere.